plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -206,7 +201,34 @@ func (c *CloudWatch) initializeCloudWatch() error {
 			},
 			Timeout: time.Duration(c.Timeout),
 		}
-	})
+	}
+
+	awsCreds, err := c.CredentialConfig.Credentials()
+	if err != nil {
+		return err
+	}
+
+	if c.CredentialConfig.EndpointURL == "" || c.CredentialConfig.Region == "" {
+		c.client = cwClient.NewFromConfig(awsCreds, clientOptions)
+	} else {
+		customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+			return aws.Endpoint{
+				PartitionID:   "aws",
+				URL:           c.CredentialConfig.EndpointURL,
+				SigningRegion: c.CredentialConfig.Region,
+			}, nil
+		})
+
+		cfg, err := awsConfig.LoadDefaultConfig(context.TODO(), awsConfig.WithEndpointResolverWithOptions(customResolver))
+		if err != nil {
+			return fmt.Errorf("unable to load AWS SDK config: " + err.Error())
+		}
```

Comment:
Can we maybe use `EndpointResolverFromURL(url string, optFns ...func(*aws.Endpoint))` here? See https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudwatch#EndpointResolverFromURL...